### DISCUSSION TITLE
Here is Gas limit issues. Tell about how to fix this issues.

### DISCUSSION THREAD

**Discussion by SuperDev313** [community/community#124385] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

```
// Function that consumes a lot of gas
contract GasHeavy {
    uint[] public data;

    function addData(uint _value) public {
        for (uint i = 0; i < 1000; i++) {
            data.push(_value);
        }
    }
}

```

**Comment by ayoub-laaouaoucha** [comment]
Instead of trying to push 1000 values in a single transaction, break the operation into smaller batches that can be executed in multiple transactions. This approach will distribute the gas cost across several transactions. 

`contract GasHeavy {
    uint[] public data;

    function addData(uint _value, uint _iterations) public {
        require(_iterations <= 100, "Too many iterations to avoid gas limit issues");
        for (uint i = 0; i < _iterations; i++) {
            data.push(_value);
        }
    }
}
`

**Comment by wnqqnw19** [comment]
This function in your contract may consume too much gas, causing it to fail.
So you can optimize the function to use less gas or split the function into smaller parts. You can do like this: 

```
contract GasOptimized {
    uint[] public data;

    function addData(uint _value, uint _iterations) public {
        for (uint i = 0; i < _iterations; i++) {
            data.push(_value);
        }
    }
}
```

**Reply by SuperDev313** [reply]
Great!

**Comment by ebndev** [comment]
Hi @SuperDev313 & @huska0707, thanks for being a part of the GitHub Community!

Please note: We made the decision to disable the ability to earn Achievements in our Community in order to discourage users from participating in coordinated or inauthentic activity like rapid questions and answers in order to earn badges. You can learn more about this decision in our announcement post here [Achievements will no longer be available in the Community](https://github.com/orgs/community/discussions/106536).

Note that GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this, and other related posts. 

Any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
